packages/webdriverio/src/middlewares.js

```diff
@@ -32,7 +33,19 @@ export const elementErrorHandler = (fn) => (commandName, commandFn) => {
                     () => {
                         return this.parent.$(this.selector).then((elem) => {
                             this.elementId = elem.elementId
-                            return fn(commandName, commandFn).apply(this, args)
+                            try{
+                                return fn(commandName, commandFn).apply(this, args)
+                            } catch(error) {
+                                if (error.message.includes("stale element reference")) {
+                                    refetchElement.call(this).then(element => {
```

Comment:
`return` is missing, otherwise we would continue without having the stale element refetched

Reply:
for better unit-testability I would call it `refetchElement(this)` so in the test you can just insert the mock as param

Reply:
Updated